dit/dit.py

```diff
@@ -453,31 +449,44 @@ def _conclude(self, data):
     # ===========================================
     # Parsers
 
-    def _gid_parse(self, argv):
+    def _gid_parser(self, argv):
         ids = argv.pop(0).split(self.separator)
         if len(ids) > 3:
             raise DitException("Invalid gID format")
-        ids = [int(i) if i else "0" for i in ids]
         ids = ids + [None] * (3 - len(ids))
         group_id, subgroup_id, task_id = ids
 
+        n_groups = len(self.index)
+        if not (group_id and group_id.isdigit()) or n_groups <= int(group_id):
+            raise DitException("Invalid group: %s" % group_id)
+        group_id = int(group_id)
```

Comment:
This same pattern seems to be used in many other places. Make a function and reuse code.

Also, try to convert to int and catch the exception. Reraise as "bad id format" or something.

It's not "Invalid group", it's "No group exists with this id"

```
try:
    group_id = int(group_id)
    group = self.index[group_id][0]
except "Bad format":
    raise BadFormat(...)
except "Wrong index":
    raise NoGroupWithId(...)